File: Programmers/1/159994. 카드 뭉치/카드 뭉치.java
import java.util.*;

class Solution {
    public String solution(String[] cards1, String[] cards2, String[] goal) {
        int p1 = 0;
        int p2 = 0;
        
        for (String str : goal) {            
            if (str.equals(cards1[p1])) {
                if (p1 < cards1.length - 1) {
                    p1++;
                }
                
                continue;
            }
            
            if (str.equals(cards2[p2])) {
                if (p2 < cards2.length - 1) {
                    p2++;
                }
                
                continue;
            }
            
            return "No";
        }
        
        return "Yes";
    }
}

File: Programmers/1/159994. 카드 뭉치/카드 뭉치.cpp
#include <string>
#include <vector>

using namespace std;

string solution(vector<string> cards1, vector<string> cards2, vector<string> goal) {
    int p1 = 0;
    int p2 = 0;
    
    for (string str : goal) {
        if (p1 < cards1.size() && str == cards1[p1])
        {
            p1++;
            continue;
        } 
        
        if (p2 < cards2.size() && str == cards2[p2])
        {
            p2++;
            continue;
        }
        
        return "No";
    }
    
    return "Yes";
}